### ISSUE TITLE
Layout/IndentationWidth picks the wrong fix

### ISSUE THREAD

**Issue by akimd** [rubocop/rubocop#14978]
Hi team!

## Steps to reproduce the problem

In the following file, the problem is that the brace is not where it should be:
```ruby
# frozen_string_literal: true

def fun
  root
    .to_h { |f|
      [f.basename.to_s, f.read]
  }
end
```

## Expected behavior

Fix the indentation of the brace:
```ruby
# frozen_string_literal: true

def fun
  root
    .to_h { |f|
      [f.basename.to_s, f.read]
    }
end
```

## Actual behavior

It changed the indentation of the body of the block.

```
$ bundle exec rubocop -A rubocop/layout_indentation_width.rb
Inspecting 1 file
C

Offenses:

rubocop/layout_indentation_width.rb:6:3: C: [Corrected] Layout/IndentationWidth: Use 2 (not 4) spaces for indentation.
      [f.basename.to_s, f.read]
  ^^^^

1 file inspected, 1 offense detected, 1 offense corrected
```
```
$ cat rubocop/layout_indentation_width.rb
# frozen_string_literal: true

def fun
  root
    .to_h { |f|
    [f.basename.to_s, f.read]
  }
end
```

## RuboCop version

```
$ bundle exec rubocop -V
1.85.0 (using Parser 3.3.10.2, Prism 1.9.0, rubocop-ast 1.49.0, analyzing as Ruby 3.4, running on ruby 3.4.8) [arm64-darwin24]
  - rubocop-minitest 0.38.2
  - rubocop-performance 1.26.1
```

**Comment by jonas054** [comment]
This is a thing you can fix with configuration.

`Layout/IndentationWidth` aligns the line `[f.basename.to_s, f.read]` relative to the end of the block, the `}` in your case, but the way you prefer to indent the closing brace is not the default one. You need to set:

```yaml
Layout/BlockAlignment:
  EnforcedStyleAlignWith: start_of_block
``` 

On a side note, with default configuration, if you manually indent `}` to where you want it, `Layout/BlockAlignment` will allow that indentation and the line above it will be autocorrected by `Layout/IndentationWidth` accordingly.

**Comment by bbatsov** [comment]
This is the expected interplay under the default config: `IndentationWidth` anchors the block body to the closing brace, and `BlockAlignment`'s default `either` accepts the brace where it is, so neither cop considers it misplaced. `Layout/BlockAlignment: EnforcedStyleAlignWith: start_of_block` autocorrects to exactly the layout you wanted, and `Layout/IndentationWidth: EnforcedStyleAlignWith: relative_to_receiver` (added in 1.87, after this report) accepts your original snippet as-is. Closing as working as intended.
